### PR TITLE
Add cloud hosting telemetry property

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

Add a `hostedOnStrapiCloud` boolean property to the tracking events sent by Strapi. 

### Why is it needed?

Monitor usage of Strapi application hosted on Strapi Cloud

### How to test it?

- `STRAPI_HOSTING=strapi.cloud yarn develop`
- Check that the `track` requests' payload includes the `hostedOnStrapiCloud=true` property 
- Intercept the backend events by modifying your `/etc/hosts` configuration and replacing `https://analytics.strapi.io` with a logger server of your choice. Alternatively, you can modify the source code of the [sender](packages/core/strapi/lib/services/metrics/sender.js) to log the anonymous metadata before they're sent.

### CODE REVIEW

File: packages/core/strapi/lib/services/metrics/sender.js [hunk 1]
```diff
@@ -54,6 +55,7 @@ module.exports = (strapi) => {
     projectType: isEE ? 'Enterprise' : 'Community',
     useTypescriptOnServer: isUsingTypeScriptSync(serverRootPath),
     useTypescriptOnAdmin: isUsingTypeScriptSync(adminRootPath),
+    hostedOnStrapiCloud: env('STRAPI_HOSTING', null) === 'StrapiCloud',
```
Comment (JabStrapi) [inline packages/core/strapi/lib/services/metrics/sender.js:58]: isHostedOnStrapiCloud
STRAPI_CLOUD=true

File: packages/generators/app/lib/utils/usage.js [hunk 2]
```diff
@@ -65,6 +65,7 @@ const getProperties = (scope, error) => ({
   useYarn: scope.useYarn,
   useTypescriptOnServer: scope.useTypescript,
   useTypescriptOnAdmin: scope.useTypescript,
+  hostedOnStrapiCloud: process.env.STRAPI_HOSTING === 'StrapiCloud',
```
Comment (JabStrapi) [inline packages/generators/app/lib/utils/usage.js:68]: isHostedOnStrapiCloud
STRAPI_CLOUD as a boolean.

**Review by JabStrapi — COMMENTED** [review]
If possible, can you apply those changes @Convly ? Thanks a lot. 
isHostedOnStrapiCloud 
STRAPI_CLOUD=True

File: packages/core/admin/server/controllers/admin.js [hunk 3]
```diff
@@ -93,11 +94,13 @@ module.exports = {
     const useTypescriptOnAdmin = await isUsingTypeScript(
       path.join(strapi.dirs.app.root, 'src', 'admin')
     );
+    const isHostedOnStrapiCloud = env.bool('STRAPI_CLOUD');
```
Comment (Elhebert) [inline packages/core/admin/server/controllers/admin.js:97]: I don't like having Cloud specific code being added to Strapi. We've always said that we should do this.
I know that this is a temporary solution until the Cloud can do its own telemetry. But still. I feel weird about this.

Comment (Convly) [reply packages/core/admin/server/controllers/admin.js]: I've reverted to `STRAPI_HOSTING` and used `strapi.cloud` as a value

**Review by Elhebert — APPROVED** [review]

**Pull request merged**